frontend/server/src/Controllers/Course.php

```diff
@@ -2058,7 +2058,18 @@ public static function apiStudentProgress(\OmegaUp\Request $r): array {
                 'assignmentNotFound'
             );
         }
+        return [
+            'problems' => self::getProblemsBySelectedAssignment(
+                $assignment,
+                $resolvedIdentity
+            ),
+        ];
+    }
 
+    private static function getProblemsBySelectedAssignment(
```

Comment:
Aquí se está quejando psalm de que no estás declarando el tipo de retorno de esta función, así que habrá que agregarla:
```suggestion
    /**
     * @return list<CourseProblem>
     */
    private static function getProblemsBySelectedAssignment(
```